### DISCUSSION TITLE
How can I escape characters using sed and kitty?

### DISCUSSION THREAD

**Discussion by SoyDiego** [kovidgoyal/kitty#7082] [Q&A]
Hi @kovidgoyal. I would like  to automatize some code and I would like when I execute my bash script on KITTY, split in 2 panes more.

I did that with the next code:

```bash
cd ~/projects/Project1 && ng s -o &

kitty @ launch sh -c "cd ~/projects/Project2 && npm run dev"
kitty @ launch sh -c "cd ~/projects/Project && npm run dev"
```

Then I wanted to complicated a little bit more and change a connection string of a file.
My connection string looks like this:

```
Server=.\\SQLEXPRESS;Database=Testing;Trusted_Connection=True;MultipleActiveResultSets=true;TrustServerCertificate=True;
```

and I want to replace it for something like this:
```
Data Source=localhost;Database=Testing123;User Id=sa;Password=MyPassword/*-432;MultipleActiveResultSets=true;Encrypt=false;
```

I did this code with `sed` and works. If I put directly in the terminal (it's replacing the line)

```bash
sed -i 's/Server=\.\\\\SQLEXPRESS;Database=Testing;Trusted_Connection=True;MultipleActiveResultSets=true;TrustServerCertificate=True;/Data Source=localhost;Database=Testing123;User Id=sa;Password=MyPassword\/*-432;MultipleActiveResultSets=true;Encrypt=false;/g' ~/projects/Project1/appsettings.json
```

My problem is when I'm trying to use with Kitty, I'm sending all the comannds but I cannot make it works. Something like this:

```bash
kitty @ launch sh -c "sed -i 's/Server=\.\\\\SQLEXPRESS;Database=Testing;Trusted_Connection=True;MultipleActiveResultSets=true;TrustServerCertificate=True;/Data Source=localhost;Database=Testing123;User Id=sa;Password=MyPassword\/*-432;MultipleActiveResultSets=true;Encrypt=false;/g' ~/projects/Project1/appsettings.json && cd ~/projects/primereact && npm run dev" 

```

Any help?
The final goal is execute the script and open in the same terminal splitted in 3 but in one of them, replace the connecting string.

I hope be clear, thanks for your time!

**Comment by cybardev** [comment]
Are you using a `startup_session` file?

If not, try that. I feel like that would be much easier to manipulate instead of having a bunch of `kitty @ launch` commands in a script.

Here's the relevant docs: https://sw.kovidgoyal.net/kitty/overview/#startup-sessions
